build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/GenerateSupportedPlatforms.cs

```diff
@@ -0,0 +1,70 @@
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	/// <summary>
+	/// Generates Microsoft.Android.Sdk.SupportedPlatforms.props
+	/// Similar to: https://github.com/dotnet/sdk/blob/18ee4eac8b3abe6d554d2e0c39d8952da0f23ce5/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.WindowsSupportedTargetPlatforms.props
+	/// </summary>
+	public class GenerateSupportedPlatforms : Task
+	{
+		/// <summary>
+		/// A list of AndroidApiInfo.xml files produced by Mono.Android.targets
+		/// </summary>
+		[Required]
+		public string [] AndroidApiInfo { get; set; }
+
+		/// <summary>
+		/// The output file to generate
+		/// </summary>
+		[Required]
+		public string OutputFile { get; set; }
+
+		public override bool Execute ()
+		{
+			var versions = new AndroidVersions (
+				AndroidApiInfo.Select (d => Path.GetDirectoryName (d)));
+			var settings = new XmlWriterSettings {
+				OmitXmlDeclaration = true,
+				Indent = true,
+			};
+			using (var writer = XmlWriter.Create (OutputFile, settings)) {
+
+				writer.WriteComment (@"
+***********************************************************************************************
+Microsoft.Android.Sdk.SupportedPlatforms.props
+
+Specifies the supported Android platform versions for this SDK.
+
+***********************************************************************************************
+");
+				writer.WriteStartElement ("Project");
+
+				writer.WriteStartElement ("PropertyGroup");
+				writer.WriteStartElement ("TargetPlatformVersion");
+				writer.WriteAttributeString ("Condition", " '$(TargetPlatformVersion)' == '' ");
+				writer.WriteString (versions.MaxStableVersion.ApiLevel.ToString ());
+				writer.WriteEndElement (); // </TargetPlatformVersion>
+				writer.WriteEndElement (); // </PropertyGroup>
+
+				writer.WriteStartElement ("ItemGroup");
+				foreach (AndroidVersion version in versions.InstalledBindingVersions) {
+					writer.WriteStartElement ("AndroidSupportedTargetPlatform");
+					writer.WriteAttributeString ("Include", version.ApiLevel.ToString ());
```

Comment:
This is supposed to be the API level and not the Android OS version?

Would list 29, 30 and not 10.0, 11.0?

Reply:
Is this perhaps an opportunity to discuss where we use API level vs OS version?  Though we need to keep `__ANDROID_29__` and such for existing code...

Reply:
I think we were planning to still use the API level, though I don't see that specifically mentioned in the specs (https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md#os-versions, https://github.com/dotnet/designs/blob/master/accepted/2020/minimum-os-version/minimum-os-version.md).

@mhutch, can you comment?

Reply:
The current generated file is:
```xml
<Project>
  <PropertyGroup>
    <TargetPlatformVersion Condition=" '$(TargetPlatformVersion)' == '' ">30</TargetPlatformVersion>
  </PropertyGroup>
  <ItemGroup>
    <AndroidSupportedTargetPlatform Include="26" />
    <AndroidSupportedTargetPlatform Include="23" />
    <AndroidSupportedTargetPlatform Include="19" />
    <AndroidSupportedTargetPlatform Include="27" />
    <AndroidSupportedTargetPlatform Include="22" />
    <AndroidSupportedTargetPlatform Include="21" />
    <AndroidSupportedTargetPlatform Include="30" />
    <AndroidSupportedTargetPlatform Include="29" />
    <AndroidSupportedTargetPlatform Include="25" />
    <AndroidSupportedTargetPlatform Include="20" />
    <AndroidSupportedTargetPlatform Include="28" />
    <AndroidSupportedTargetPlatform Include="24" />
    <SupportedTargetPlatform Condition=" '$(TargetPlatformIdentifier)' == 'Android' " Include="@(AndroidSupportedTargetPlatform)" />
  </ItemGroup>
</Project>
```

I need to drop 19 & 20, and looks like they need to be sorted, too. 🤦‍♀️

Reply:
Latest file is:
```xml
<Project>
  <PropertyGroup>
    <TargetPlatformVersion Condition=" '$(TargetPlatformVersion)' == '' ">30</TargetPlatformVersion>
  </PropertyGroup>
  <ItemGroup>
    <AndroidSupportedTargetPlatform Include="21" />
    <AndroidSupportedTargetPlatform Include="22" />
    <AndroidSupportedTargetPlatform Include="23" />
    <AndroidSupportedTargetPlatform Include="24" />
    <AndroidSupportedTargetPlatform Include="25" />
    <AndroidSupportedTargetPlatform Include="26" />
    <AndroidSupportedTargetPlatform Include="27" />
    <AndroidSupportedTargetPlatform Include="28" />
    <AndroidSupportedTargetPlatform Include="29" />
    <AndroidSupportedTargetPlatform Include="30" />
    <SupportedTargetPlatform Condition=" '$(TargetPlatformIdentifier)' == 'Android' " Include="@(AndroidSupportedTargetPlatform)" />
  </ItemGroup>
</Project>
```

Reply:
@dsplaisted I filed an issue on the dotnet/designs repo a while back about whether we want to use the API level or OS version for android TFMs: https://github.com/dotnet/designs/issues/136.

It doesn't seem like there was a conclusion but I agree this is something that needs to be decided sooner than later.